go.php

```diff
@@ -82,13 +82,49 @@
 
 header('HTTP/1.1 302 Moved Temporarily');
 if (array_key_exists($from, $mapping)) {
-    header('Location: ' . $location . $mapping[$from]);
+    $subPath = $mapping[$from];
 } else {
     if (strpos($from, 'admin-') === 0) {
-        header('Location: ' . $location . '/admin_manual');
+        $subPath = '/admin_manual';
     } else if (strpos($from, 'developer-') === 0) {
-        header('Location: ' . $location . '/developer_manual');
+        $subPath = '/developer_manual';
     } else {
-        header('Location: ' . $location . '/user_manual');
+        $subPath = '/user_manual';
     }
 }
+
+$subPathParts = explode('/', $subPath);
+$manual = $subPathParts[1];
+if ($manual === 'user_manual') {
+    // Sort accepted languages by their weight
+    $acceptLanguages = array_reduce(
+        explode(', ', $_SERVER['HTTP_ACCEPT_LANGUAGE']),
+        static function ($out, $element) {
+            [$language, $q] = array_merge(explode(';q=', $element), [1]);
+
+            $out[$language] = (float)$q;
+
+            return $out;
+        },
+        [],
+    );
+    arsort($acceptLanguages);
+
+    foreach ($acceptLanguages as $language => $weight) {
+        if (!preg_match('/^[a-z-]+$/im', $language)) {
+            // Skip any invalid languages and prevent directory traversals
+            continue;
+        }
+
+        $language = str_replace('-', '_', $language);
+
+        // To test locally add '/_build/html/' to the path.
+        if (file_exists(__DIR__ . '/user_manual/' . $language)) {
+            // Insert the language /user_manual/<language>/...
+            $subPath = implode('/', array_merge(array_slice($subPathParts, 0, 2), [$language], array_slice($subPathParts, 2)));
```

Comment:
```suggestion
            array_splice($subPathParts, 2, 0, [$language]);
            $subPath = implode('/', $subPathParts);
```
Not sure if prettier. When I started writing I thought array_splice returns the result but it actually modifies and return the remove elements, so `[]` here, so we need 2 lines.

Reply:
It's easier to understand, so I'll take it :+1: